hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieInstantTimeGenerator.java

```diff
@@ -144,4 +144,10 @@ public static boolean isValidInstantTime(String instantTime) {
       return false;
     }
   }
+
+  private static ZoneId getZoneId() {
+    return commitTimeZone.equals(HoodieTimelineTimeZone.LOCAL)
+        ? ZoneId.systemDefault()
```

Comment:
If possible, fetch the timezone whout metaClient.tableConfig, the `HoodieTimelineTimeZone` can not assure the initialization of zoneId.

Reply:
> If possible, fetch the timezone whout metaClient.tableConfig, the `HoodieTimelineTimeZone` can not assure the initialization of zoneId.

 I will try to modify the code as you say

Reply:
> If possible, fetch the timezone whout metaClient.tableConfig, the `HoodieTimelineTimeZone` can not assure the initialization of zoneId.


In the class `HoodieInstantTimeGenerator`, set an initial value( `HoodieTimelineTimeZone.LOCAL` ) for the property `commitTimeZone` 
```java
private static HoodieTimelineTimeZone commitTimeZone = HoodieTimelineTimeZone.LOCAL;
```
And update `commitTimeZone` value in `HoodieTableConfig#create`
```java
if (hoodieConfig.contains(TIMELINE_TIMEZONE)) {
    HoodieInstantTimeGenerator.setCommitTimeZone(HoodieTimelineTimeZone.valueOf(hoodieConfig.getString(TIMELINE_TIMEZONE)));
}
```

```java
public static void setCommitTimeZone(HoodieTimelineTimeZone commitTimeZone) {
  HoodieInstantTimeGenerator.commitTimeZone = commitTimeZone;
}
```
So, I think getting ZoneId by HoodieTimelineTimeZone should be correct. and I don't really understand the meaning of `the HoodieTimelineTimeZone can not assure the initialization of zoneId`.
I don't know if my idea is correct, looking forward to your reply.

Reply:
See the discussions we take in: https://github.com/apache/hudi/pull/8631

Reply:
> metaClient

I sees, I will try to modify the code as you say.

Reply:
> See the discussions we take in: #8631

I sees, I will try to modify the code as you say.

Reply:
> See the discussions we take in: #8631

It seems that there is no good way to get `HoodieTimelineTimeZone` through `HoodieTableMetaClient` in `HoodieInstantTimeGenerator`, I currently get `HoodieTimelineTimeZone` by instantiate a `HoodieTableConfig`, can you give me some advice?

Reply:
> I currently get HoodieTimelineTimeZone by instantiate a HoodieTableConfig

If no existing table meta client or table config can be reused, we must instantiate a new one.